rolodex/forecast/forecast_index.py

```diff
@@ -203,14 +204,29 @@ def get_indexer(
         else:
             nsteps = period_index.size
 
+        time_diff = np.diff(time_index)
+        period_diff = np.diff(period_index)
+        
+        # assume that the period differences are constant 
+        n_best_steps_per_forecast = (time_diff / period_diff[0]).astype(int) 
+
         needed_time_idxrs = np.concatenate(
             [
-                np.arange(first_index, last_index, dtype=int),
-                np.repeat(last_index, nsteps),
+                np.repeat(i, n_best_steps_per_forecast[i])
+                for i in range(len(time_index)- 1)
```

Comment:
Think this should work.

```suggestion
                np.repeat(np.arange(len(time_index)- 1), n_best_steps_per_forecast)
```

Since
```
>>> np.repeat([1, 2, 3], [3, 2, 1])
array([1, 1, 1, 2, 2, 3])
```